presto-rcfile/src/main/java/com/facebook/presto/rcfile/BufferedOutputStreamSliceOutput.java

```diff
@@ -173,17 +173,31 @@ public void writeBytes(Slice source)
     @Override
     public void writeBytes(Slice source, int sourceIndex, int length)
     {
-        // Write huge chunks direct to OutputStream
-        if (length >= MINIMUM_CHUNK_SIZE) {
-            flushBufferToOutputStream();
-            writeToOutputStream(source, sourceIndex, length);
-            bufferOffset += length;
-        }
-        else {
-            ensureWritableBytes(length);
-            slice.setBytes(bufferPosition, source, sourceIndex, length);
-            bufferPosition += length;
+        if (length >= slice.length()) {
+            if (bufferPosition > 0) {
+                // fill up the current buffer
+                int flushLength = getFreeBufferLength();
+                slice.setBytes(bufferPosition, source, sourceIndex, flushLength);
+                bufferPosition = slice.length();
+                flushBufferToOutputStream();
+                sourceIndex += flushLength;
+                length -= flushLength;
+            }
+
+            // line up the chunk to chunk size and flush directly to OutputStream
+            int flushLength = length - length % CHUNK_SIZE;
+            for (int i = 0; i < flushLength / CHUNK_SIZE; i++) {
```

Comment:
I think it would be simpler to do:
```java
while (length <= CHUNK_SIZE) {
    writeToOutputStream(source, sourceIndex, CHUNK_SIZE);
    sourceIndex += CHUNK_SIZE;
    length -= CHUNK_SIZE;
}
```
then after the loop you don't need to touch length of bufferOffset.